src/server/node_manager.c

```diff
@@ -7130,10 +7130,13 @@ set_nodes(void *pobj, int objtype, char *execvnod_in, char **execvnod_out, char
 				return (PBSE_UNKNODE);
 			}
 
+			if ((pnode->nd_state & (INUSE_DOWN | INUSE_STALE | INUSE_OFFLINE)) && (svr_init == FALSE))
+				if ((objtype == RESC_RESV_OBJECT) && (presv->ri_qs.ri_resvID[0] != PBS_MNTNC_RESV_ID_CHAR) /*&& (presv->ri_qs.ri_state == RESV_UNCONFIRMED)*/)
+					set_resv_for_degrade(pnode, presv);
```

Comment:
There is a chunk of code in this function where we check the same thing for jobs and reservations.  Please combine these two checks.

Reply:
I am unable to find the chunk where we check if the node is (down, stale or offline) for a job in this function.

Reply:
Oh you are right.  I added it, then removed it due to all the problems it was causing.  The code worked for both jobs and reservations.  Reservations were the problematic code (which you've correctly handled).  We should probably submit a ticket to handle jobs in this fashion too.  If a mom is down, we still try to send the job to the mom even though we know it is down.

Reply:
I am of the view that we need not handle the job case. The node can go down right after we check if it is down. So, it is better that we fail at the time of sending the job to the mom, which I think is very soon after we assign nodes to the job.
Also, if we reject the request on finding the node is down, the scheduler is not going to look for alternate nodes. What do you think?

Reply:
If we know the node is down, why should we go through the extra work to try and send it to the mom just to fail?  We know it will fail.  If the node goes down after set_nodes(), we'll still catch it like we do now.  As for the scheduler finding a new node, this will only happen in the case where throughput mode is false.